src/quic.c

```diff
@@ -554,6 +554,37 @@ void wolfSSL_set_quic_early_data_enabled(WOLFSSL* ssl, int enabled)
 }
 #endif /* WOLFSSL_EARLY_DATA */
 
+int wolfSSL_quic_do_handshake(WOLFSSL* ssl)
+{
+    int ret = WOLFSSL_SUCCESS;
+
+    WOLFSSL_ENTER("wolfSSL_quic_do_handshake");
+
+    if (!wolfSSL_is_quic(ssl)) {
```

Comment:
I would prefer such a simple function to avoid using `goto`, however we do this in many places.

Reply:
A matter of taste, definitely. I found that a single function exit increases ease of maintenance drastically. 

Everyone who introduces a new "resource holder" (e.g. a memory alloc) can add deallocation in one place and does not have to care about early returns. Similarly, one can add early returns (goto cleanup;) without need to understand all states of allocated resources.